generic.go

```diff
@@ -95,21 +74,75 @@ func NewGeneric(config *Config, queue workqueue.RateLimitingInterface, workers i
 				Help:      fmt.Sprintf("Histogram measuring the time it took to process a %s", &groupKind),
 			},
 		)
+		// Extra api data
+		requestCount := prometheus.NewCounterVec(
```

Comment:
These (and above) should only be constructed and registered IF there is a controller and/or a server that this constructor wants us to start.

Reply:
For the controller that makes sense but we need them before the server is created to setup the middleware functionality.

Reply:
Right. Then construct them but do not register unless there is a server actually?